packages/extension-sample/src/setup-app.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+import loadable from '@loadable/component'
+import {IApplicationExtension, IRouteConfig} from '@salesforce/pwa-kit-react-sdk/ssr/universal/extensibility/types'
+import {IConfig} from './types'
+import withRedBorder from './components/with-red-border'
+
+const SamplePage = loadable(() => import('./pages/sample'))
+
+const defaultPath: string = '/sample-page'
+class Sample implements IApplicationExtension {
```

Comment:
This a a pattern I followed from @kevinxh's work. "app" in setup-app is referring to the react application being setup.. not the setting up of the extensions. This is the same for setup-server where server is the expressjs application.

Reply:
I personally don't think that these will be the final names for the files that we end up using, but I think it's going to go through a few iterations. I don't want to slow down development for naming conversations, we can take that to our weekly meetings.

Reply:
Yeah that's all right. We don't need to finalize the names in this PR. I'll add a topic to discuss them in the weekly meeting. 

While I was reviewing the code, I'm actually more like observing and learning. So the naming is important to me. It's kinda confusing that the word `app` seems to be used interchangeably: sometimes it means extending the general app/project, while other times it could mean specifically extending the React App component.

Reply:
Might be something that we can think about adding to the glossary section of the team doc once we agree on things